projects/plugins/crm/includes/ZeroBSCRM.DAL3.Obj.Companies.php

```diff
@@ -2759,8 +2761,22 @@ private function tidy_company($obj=false,$withCustomFields=false){
 
             // if have totals, pass them :)
             if (isset($obj->quotes_total)) $res['quotes_total'] = $obj->quotes_total;
-            if (isset($obj->invoices_total)) $res['invoices_total'] = $obj->invoices_total;
-            if (isset($obj->transactions_total)) $res['transactions_total'] = $obj->transactions_total;
+			if ( isset( $obj->invoices_total ) ) {
+				$company                            = $zbs->DAL->companies->getCompany( $res['id'], array( 'withInvoices' => true ) ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				$deleted_invoice_details            = jetpackCRM_deleted_totals( $company['invoices'] );
+				$res['invoices_total']              = $obj->invoices_total - $deleted_invoice_details['total'];
+				$res['invoices_total_with_deleted'] = $deleted_invoice_details['total'];
+				$res['invoices_count_with_deleted'] = $zbs->DAL->customer_has_count_obj_type( $res['id'], ZBS_TYPE_INVOICE, ZBS_TYPE_COMPANY ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
```

Comment:
Perhaps simpler (if I'm understanding correctly):

```suggestion
				$res['invoices_count_with_deleted'] = $company['invoices'];
```

Reply:
> $res['invoices_count_with_deleted'] = count( $company['invoices'] );

Yes, thank you, that would work. It is simpler, with no DB query (added it to a new commit).

>  count( $obj->invoices );

The $obj object doesn't return the list of invoices so that doesn't work.